java-generator/core/src/test/java/io/fabric8/java/generator/GeneratorTest.java

```diff
@@ -504,4 +511,60 @@ void testClassNamesDisambiguationWithPackageNesting() {
     assertTrue(clzO2.get().getFieldByName("o2").isPresent());
     assertTrue(clzO2.get().getFieldByName("o3").isPresent());
   }
+
+  @Test
+  void testObjectWithDefaultFields() {
```

Comment:
this test looks pretty complicated, as we already have compilation and approval tests to verify the actual "content" of the default, I would be happy to only test that a default value exists/doesn't exists when is defined/when is not.

I would be actually more interested into using the integration tests with the Maven plugin to verify that the behaviour is according to specification e.g.: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable

Reply:
Test is now way more simple. Integration testing can be done in a different PR.